middleman-core/fixtures/traversal-app/config.rb

```diff
@@ -1,3 +1,30 @@
+class MyFeature < Middleman::Extension
+
+  def manipulate_resource_list(resources)
+
+    resources.each do |resource|
+
+      # Contrived example: we're removing integer prefixes from file names
+      # during output, e.g., rename_paths/30_item_three/10_sibling_one.html.md
+      # becomes rename_paths/item_three/sibling_one.html
+
+      if ( match = /(\d+?)_/.match(resource.path) )
+        resource.destination_path = resource.destination_path.gsub(/(\d+?)_/, '') # fails!
+        puts resource.destination_path
+        # resource.destination_path.gsub!(/(\d+?)_/, '') # works!
```

Comment:
That bit changes the output path of various items, e.g. `rename_paths/30_item_three/20_sibling_two.html` becomes `rename_paths/item_three/sibling_two.html` on output.

By "contrived example" I mean a real application would be a bit more sophisticated, but an example might be using page numbers or articles dates.